### PR TITLE
Redirect to interactive based on spreadsheet

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
https://docs.google.com/spreadsheets/d/1Rlx_BfkwCpE8KBJZVK_p5gO61HZF7fRckQige6Hwbl8/edit#gid=0

Also gives the option of embedding the iframe directly in the 'next' page

### CODE REVIEW

File: server/controllers/interactive.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = function (req, res, next) {
+	// assumes polling has already started
+	var wasRedirected = require('../lib/ig-poller').getData().some(function (mapping) {
+		if (mapping.articleuuid === req.params.id) {
+			var interactiveFullUrl = 'http://ig.ft.com' + mapping.interactiveurl;
```
Comment (matthew-andrews) [inline server/controllers/interactive.js]: should we limit only to `ig.ft.com` domains?

**Comment by matthew-andrews** [comment]
:+1:

**Comment by matthew-andrews** [comment]
@kavanagh should have a play with this :-)

File: server/controllers/interactive.js (hunk 1 above)
Comment (ironsidevsquincy) [inline server/controllers/interactive.js]: if we don't, people would be opted out, no?

Comment (matthew-andrews) [inline server/controllers/interactive.js]: i was just thinking out loud whether we should allow any website to be 302'd/proxy'd. but even the ft.com IG links wouldn't opt people out  they would just get bounced about, e.g:-

http://next.ft.com/__opt-in
http://www.ft.com/ig/sites/2015/special-reports/shanghai/

**Comment by matthew-andrews** [comment]
i've tried to improve the nightwatch tests so hopefully wont see this failure again

**Comment by kavanagh** [comment]
Wait, let me get up to date with this.... this is what I think you've done:

1. when we create link files in Methode we also enter the UUID into this spreadsheet (because link files are not a thing in CAPI)
2. when you get a request for a UUID that's not in CAPI you then lookup to see if that's a UUID in the spreadsheet
3. You then either redirect to the URL the browser of embed it with an iframe filling the page.

Is that correct?

Sounds good.

What's the rationale behind the iframe bit?

**Comment by kavanagh** [comment]
ps I don't have access to the spreadsheet so trying to figure it our based on the Bertha response.

**Comment by matthew-andrews** [comment]
Yes the workflow above would work but the intention is for it to be more versatile than that…

- You could, if you wanted, author the content that would be in the interactive graphic in Methode *properly* so that it *does* come through the API.  Then the IG itself could be responsible for loading & displaying that content.  This would also mean all the content in microsites built in this way would be available in search and have proper metadata associated with it such that the underlying articles would appear on stream pages but they would never render as standard articles on Next.
- Then for each article of the type above you would enter the UUIDs into the spreadsheet (which you now have access to).  If the IG supported `#` anchors you could even have one article per chapter, say, and link directly to it.
- The rational for the iframe is simply that we want to provide as much flexibility as possible……  Once this PR is merged you would have two options — either making those articles 302 to the IG or proxying through to Next via an iframe.  In the future we could even proxy directly onto Next on the server side.  (This would require IG's to have absolute paths to its assets or some magic on our side)

**Comment by matthew-andrews** [comment]
Ultimately the goal is enable Next to help IG solve the problem that the content within Interactive Graphics don't appear on Next's metadata-driven stream pages (to come through, the content *needs* to be real Content API objects — not link files); don't work with Next's follow and don't appear in search and in general don't work like other articles on FT.com.  (The proposal is making work like normal articles simply by making them actual FT articles in every way except for render)

File: server/controllers/interactive.js (hunk 1 above)
Comment (ironsidevsquincy) [inline server/controllers/interactive.js]: good point. I'll change so the full url is entered into the spreadsheet, more flexible

**Pull request merged**